lib/Doctrine/DBAL/Types/Type.php

```diff
@@ -159,19 +159,10 @@ public function getDefaultLength(AbstractPlatform $platform)
     abstract public function getSQLDeclaration(array $fieldDeclaration, AbstractPlatform $platform);
 
     /**
-     * Gets the name of this type.
-     *
-     * @return string
-     *
-     * @todo Needed?
-     */
-    abstract public function getName();
```

Comment:
Please do deprecate it instead.

Also, it is not useless: consumers of `Type` objects do need to get the name eventually, although the class name is indeed sufficient once we replace custom names in mappings.

Until then, this should stay.

Reply:
Ok, so to be clear on what I'm supposed to do:

1. change the base on 2.5
2. add an `@deprecated` annotation
3. implement the method like this:
```php
<?php
@trigger_error(
    'This method is deprecated, please use the class name directly instead',
    E_USER_DEPRECATED
);
```